simulator/include/sim/simulator.hpp

```diff
@@ -4,21 +4,30 @@
 #include <sim/common.hpp>
 #include <sim/hart.hpp>
 #include <sim/memory.hpp>
+#include <sim/instr.hpp>
 
 namespace sim {
 
-struct Simulator final {
+class Simulator final {
```

Comment:
It was written in accordance with project codestyle: without excessive access specifiers. Revert this, please

Reply:
fixed